middleware/id_validation.go

```diff
@@ -0,0 +1,72 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/RedHatInsights/sources-api-go/util"
+	"github.com/labstack/echo/v4"
+)
+
+// IdValidation takes all the parameters which end with "id" and checks for their validity. Returns a bad request if
+// they're not valid.
+func IdValidation(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		for _, param := range c.ParamNames() {
+			if strings.HasSuffix(param, "id") {
+				err := validateId(c, param)
+				if err != nil {
+					return util.NewErrBadRequest(err)
+				}
+			}
+		}
+
+		return next(c)
+	}
+}
+
+// UuidValidation checks if the UUID parameter is valid. Returns a bad request if it isn't.
+func UuidValidation(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		err := validateUuid(c)
+		if err != nil {
+			return util.NewErrBadRequest(err)
+		}
+
+		return next(c)
+	}
+}
+
+// validateId checks if the ID stored in specified context parameter isn't empty, is a valid integer and that it is
+// greater than zero.
+func validateId(c echo.Context, idParamName string) error {
+	idRaw := c.Param(idParamName)
+
+	if idRaw == "" {
+		return errors.New("the provided ID cannot be empty or missing")
+	}
+
+	id, err := strconv.ParseInt(idRaw, 10, 64)
+	if err != nil {
+		return fmt.Errorf("could not parse the provided ID: %s", err)
+	}
+
+	if !(id > 0) {
+		return errors.New("the provided ID must be greater than zero")
+	}
+
+	return nil
+}
+
+// validateUuid checks if the UUID is not empty.
+func validateUuid(c echo.Context) error {
+	uid := c.Param("uid")
```

Comment:
Do need to validate here just string without special characters ?
`uid==""` is not possible to reach.

Reply:
I was thinking on sticking to the UUID specification, but I don't know why a "we might not receive UUIDs sometimes" crossed my mind, and I didn't implement it, since it might be too strict...

I'm unsure about what to do :thinking:

Reply:
OK - let's leave it as it is now I don't know either.